enhancements/monitoring/monitoring-stack-operator.md

```diff
@@ -0,0 +1,504 @@
+
+In this manner, managed service owners can simply create an instance of `MonitoringStack` alongside their services and delegate the provisioning and management of their monitoring stack to the operator. This can be done on runtime to e.g create email template for Alertmanager based on cluster data.
+
+An example `MonitoringStack` resource could look as follows:
+
+```yaml
+---
+apiVersion: monitoring.coreos.io/v1alpha1
+kind: MonitoringStack
+metadata:
+  name: monitoring
+  namespace: redhat-kafka
+spec:
+  resourceSelector:
+    matchLabels:
+      system: kafka
+  retention: 12d
+  resources:
+    requests: 
+      memory: 512M
+    limits:  
+      memory: 1G
+---
+```
+
+In the first iteration, each monitoring stack will be deployed and scoped to a single namespace. This means that the Prometheus instance will only be able to scrape monitoring resources from the namespace it is deployed in. In subsequent iterations, the CRD can be extended to allow scraping from multiple namespaces, either by namespace name or by namespace label selector.
+
+With such a design, User Workload Monitoring could become just another instance of a `MonitoringStack`. Furthermore, Platform Monitoring can be trimmed down in the same way by allowing various OpenShift core components to deploy their `MonitoringStack` resource.
+
+The following diagram illustrates a high-level overview of the final solution applied in the context of managed services.
+
+![drawing](./assets/monitoring-stack-diagram.png)
+
+_First iteration of the new monitoring operator, showing a monitoring stack with a Prometheus, Grafana and a shared Alertmanager._
+
+## Design Details
+
+### Installation
+
+The newly developed operator will be deployed using the Operator Lifecycle Manager (OLM).
+
+This will allow us to make releases much faster and iteratively and incrementally. As a result, we will be able to respond to new requirements much more rapidly. Release batches will also become smaller, reducing the risk of multiple independent features affecting each other.
+
+In order to be aligned with the descoping [plan](https://docs.google.com/presentation/d/1j1J575SxS8LtL_YvKqrexUhso7j4SgrLfyNrDUroJcc/edit#slide=id.ga089527607_0_0) that the OLM team is currently working towards, we recommend avoiding [`OperatorGroups`](https://docs.openshift.com/container-platform/4.2/operators/understanding_olm/olm-understanding-operatorgroups.html).
+There is a good chance that they will become deprecated in the future (possibly as soon as OpenShift 4.11).
+For this reason, the newly developed operator should be installed globally in a cluster and should provide its own CRDs and Roles as part of the installation bundle. A cluster administrator would then create RoleBindings to bind the roles to namespaces to which the operator needs access.
```

Comment:
> For this reason, the newly developed operator should be installed globally

Isn't conflicting with the below statement?

> In the first iteration, each monitoring stack will be deployed and scoped to a single namespace.

Reply:
> For this reason, the newly developed operator should be installed globally in a cluster and should provide its own CRDs

Does it mean the new operator has to ship Prometheus related CRDs(PromethesRule, AMConfig,..) ? If so, wouldn't cause conflict with CRDs which already shipped by CMO?